frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/OptionListEditor.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { createRef, useState } from 'react';
+import type { IGenericEditComponent } from '@altinn/ux-editor/components/config/componentConfig';
+import type { SelectionComponentType } from '@altinn/ux-editor/types/FormComponent';
+import type { Option } from 'app-shared/types/Option';
+import { useTranslation } from 'react-i18next';
+import {
+  StudioCodeListEditor,
+  StudioModal,
+  StudioSpinner,
+  StudioErrorMessage,
+} from '@studio/components';
+import { TableIcon } from '@studio/icons';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useUpdateOptionListMutation } from '../../../../../hooks/mutations/useUpdateOptionListMutation';
+import { useOptionListsQuery } from '../../../../../hooks/queries/useOptionListsQuery';
+import { useOptionListEditorTexts } from './hooks/useOptionListEditorTexts';
+import { usePreviewContext } from 'app-development/contexts/PreviewContext';
+import classes from './OptionListEditor.module.css';
+
+type OptionListEditorProps = Pick<IGenericEditComponent<SelectionComponentType>, 'component'>;
+
+export function OptionListEditor({ component }: OptionListEditorProps): React.ReactNode {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { data: optionsListMap, status } = useOptionListsQuery(org, app);
+
+  switch (status) {
+    case 'pending':
+      return (
+        <StudioSpinner spinnerTitle={t('ux_editor.modal_properties_code_list_spinner_title')} />
+      );
+    case 'error':
+      return (
+        <StudioErrorMessage>{t('ux_editor.modal_properties_error_message')}</StudioErrorMessage>
+      );
+    case 'success': {
+      return (
+        <OptionListEditorModal
+          optionList={optionsListMap.get(component.optionsId)}
+          component={component}
+        />
+      );
+    }
+  }
+}
+
+type OptionListEditorModalProps = {
+  optionList: Option[];
+} & Pick<IGenericEditComponent<SelectionComponentType>, 'component'>;
+
+function OptionListEditorModal({
+  optionList,
+  component,
+}: OptionListEditorModalProps): React.ReactNode {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { doReloadPreview } = usePreviewContext();
+  const { mutate: uploadOptionList } = useUpdateOptionListMutation(org, app);
+  const [currentOptionList, setCurrentOptionList] = useState<Option[]>(optionList);
+  const editorTexts = useOptionListEditorTexts();
+  const modalRef = createRef<HTMLDialogElement>();
+
+  const handleOptionsChange = (options: Option[]) => {
+    setCurrentOptionList(options);
+  };
+
+  const handleClose = () => {
+    uploadOptionList({ optionListId: component.optionsId, optionsList: currentOptionList });
+    doReloadPreview();
+    modalRef.current?.close();
+  };
+
+  const handleClick = () => {
+    setCurrentOptionList(optionList);
+  };
```

Comment:
And I dont think you need this either? 🤔

Reply:
If we remove the useState, then we do not need this either. See above for more context on the useState

Reply:
Will not the state be set to the incoming codeList anyway? Even if we use the state or not 🤔